src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -438,4 +464,18 @@ public Post findById(Long postId) {
         coursePosts.sort(Comparator.comparing((coursePost) -> postContentCompareStrategy.performSimilarityCheck(post, coursePost)));
         return Lists.reverse(coursePosts).stream().limit(TOP_K_SIMILARITY_RESULTS).collect(Collectors.toList());
     }
+
+    /**
+     * Checks if the requesting user is authorized in the course context,
+     * i.e., if the user is allowed to interact with a certain post
+     *
+     * @param post      post to interact with, i.e., create, update or delete
+     * @param user      requesting user
+     * @param course    course the posting belongs to
+     */
+    private void mayInteractWithPost(Post post, User user, Course course) {
```

Comment:
Should we also add the "ElseThrow" suffix to the method name?

Reply:
yes, it's more consistent, I will change it.